src/transformers/models/whisper/convert_openai_to_hf.py

```diff
@@ -125,14 +125,14 @@ def _download(url: str, root: str) -> bytes:
             "Model has been downloaded but the SHA256 checksum does not not match. Please retry loading the model."
         )
 
-    return model_bytes
+    return download_target
 
 
 def convert_openai_whisper_to_tfms(checkpoint_path, pytorch_dump_folder_path):
     if ".pt" not in checkpoint_path:
-        original_checkpoint = _download(_MODELS[checkpoint_path])
-    else:
-        original_checkpoint = torch.load(checkpoint_path, map_location="cpu")
+        root = os.path.dirname(pytorch_dump_folder_path) or "."
+        checkpoint_path = _download(_MODELS[checkpoint_path], root)
+    original_checkpoint = torch.load(checkpoint_path, map_location="cpu")
```

Comment:
actually with this we end up loading the checkpoint twice, which can be a problem for bigger models, maybe juste adding the root path can do the trick WDYT?

Reply:
Good point. We are reading the model file twice.

Before the [`_download()`](https://github.com/huggingface/transformers/blob/51042ae8e5df8275d16b3eaff36c20fb9c191655/src/transformers/models/whisper/convert_openai_to_hf.py#L93-L128) function returned raw `bytes`, not a `dict`. So [the following code](https://github.com/huggingface/transformers/blob/51042ae8e5df8275d16b3eaff36c20fb9c191655/src/transformers/models/whisper/convert_openai_to_hf.py#L131-L136) did not work:

```python
def _download(url: str, root: str) -> bytes:
    # [...]
    return model_bytes


def convert_openai_whisper_to_tfms(checkpoint_path, pytorch_dump_folder_path):
    if ".pt" not in checkpoint_path:
        original_checkpoint = _download(_MODELS[checkpoint_path])
    else:
        original_checkpoint = torch.load(checkpoint_path, map_location="CPU")
    # Here, it gave the "byte indices must be integers or slices, not str" error mentioned:
    dimensions = original_checkpoint["dims"]
    state_dict = original_checkpoint["model_state_dict"]
```

I have modified the `_download()` code to reuse the read `model_bytes` by converting it to `io.BytesIO`, and then calling `torch.load()`:

```python
def _download(url: str, root: str) -> io.BytesIO:
    # [...]
    return torch.load(io.BytesIO(model_bytes))
```
Let me know if you know or find a better approach.